### PR TITLE
Provide getTitleOrUrl function to return url if title is empty

### PR DESCRIPTION
**Pull request by Jaifroid**
This addresses #485. Is this the right way to define the function? I've tested in Chromium and Edge in jQuery mode so far, and it seems to work.

### CODE REVIEW

**Comment by Jaifroid** [comment]
Further info: this PR needs testing against a ZIM with new-style URLs, e.g. `wikivoyage_en_all_novid_2019-04.zim`. It's what I used in my quick tests so far (also now tested in SW mode on Chromium and Edge).

As an aside, note that this ZIM is buggy. For example, the Lausanne article, in SW Mode, has all the section titles closed and there is no way to open them. Also, the home page is missing, so instead we geta a huge list of all the titles in the ZIM which takes ages to load.

File: www/js/lib/zimDirEntry.js [hunk 1]
```diff
@@ -107,6 +107,15 @@ define([], function() {
         return new DirEntry(zimfile, data);
     };
 
+    /**
+     * Defines a getter function that returns the URL if the title is empty
```
Comment (mossroy) [inline www/js/lib/zimDirEntry.js]: Here, I would not say it is "getter function" (might be confusing).
And I would refer to the spec to explain why we implement this fallback

Comment (Jaifroid) [reply www/js/lib/zimDirEntry.js]: OK, will do.

**Comment by Jaifroid** [comment]
I'll take a look at these extra places. Unless the code is very obvious, I might need to leave the unit tests to you, as I've never worked with them before.

**Comment by mossroy** [comment]
OK no problem : I can handle the unit tests if you're not comfortable with them

**Comment by Jaifroid** [comment]
> OK no problem : I can handle the unit tests if you're not comfortable with them

That would be great, thanks!

**Comment by mossroy** [comment]
I commited the changes in Unit tests, and Travis seems happy with them

**Comment by Jaifroid** [comment]
Hmm, checks have failed in this latest. Maybe we pushed too many changes too quickly?

File: www/js/lib/util.js [hunk 2]
```diff
@@ -86,12 +86,12 @@ define(['q'], function(q) {
...
         array.sort(function(dirEntryA, dirEntryB) {
-            if (dirEntryA.title < dirEntryB.title) return -1;
-            if (dirEntryA.title > dirEntryB.title) return 1;
+            if (dirEntryA.getTitleOrUrl() < dirEntryB.getTitleOrUrl()) return -1;
+            if (dirEntryA.getTitleOrUrl() > dirEntryB.getTitleOrUrl()) return 1;
             return 0;
         });
         for(var i = 1; i < array.length; ){
-            if(array[i-1].title === array[i].title){
+            if(array[i-1].getTitleOrUrl() === array[i].getTitleOrUrl()){
```
Comment (Jaifroid) [inline www/js/lib/util.js]: @mossroy Is this line the problem? I assumed this was an array of dirEntries, but maybe it isn't.

Comment (mossroy) [reply www/js/lib/util.js]: Sorry, my bad : it's supposed to run on DirEntries, but the unit test does not (and it's unused in the actual code). You can leave it with .title for this PR.
Maybe we should remove this function and its unit test

**Comment by Jaifroid** [comment]
No problem, I've reverted that commit. I'm fine about removing the unused function. A different PR?

**Comment by mossroy** [comment]
It now looks good.
Yes, removing the unused function (and the corresponding unit test) would be better in a different PR

**Comment by Jaifroid** [comment]
I don't think I have any further changes for this PR. When you're happy with it (after you've done any further testing you'd like to do), please approve the PR. I'll then squash/merge.

**Comment by mossroy** [comment]
Sorry, I forgot to officially approve with the github feature. I did not test, but the code is OK for me
